static/js/components/SourceDesktop.jsx

```diff
@@ -230,8 +231,11 @@ const SourceDesktop = ({ source }) => {
   }
   const specIDs = spectra ? spectra.map((s) => s.id).join(",") : "";
 
+  const associatedGCNs = useSelector((state) => state.source.associatedGCNs);
+
   useEffect(() => {
     dispatch(spectraActions.fetchSourceSpectra(source.id));
+    dispatch(sourceActions.fetchAssociatedGCNs(source.id));
```

Comment:
We expect this is still fast enough on a DB with a lot of GCNs because its just that new table right?

Reply:
Yep! its just fetching a few rows on that new table.